packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/exporters/CompressionImageExporter.java

```diff
@@ -0,0 +1,92 @@
+package expo.modules.imagepicker.exporters;
+
+import android.graphics.Bitmap;
+import android.net.Uri;
+
+import org.apache.commons.io.FilenameUtils;
+import org.jetbrains.annotations.NotNull;
+import org.unimodules.interfaces.imageloader.ImageLoader;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+public class CompressionImageExporter implements ImageExporter {
```

Comment:
This class should also be easily testable, though I am not sure whether `android.graphics.Bitmap` manipulation won't make it painful.